packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -539,6 +542,12 @@ class DeviceDomain extends Domain {
     if (!discoverer.supportsPlatform)
       return;
 
+    if (!discoverer.canListAnything)
+      sendEvent(
+          'daemon.showMessage',
+          'Unable to discover ${discoverer.name}. Please run "flutter doctor" '
```

Comment:
(A client UI will likely have a title and message field, and it'd be nice if the daemon provided a useful title for each message).